components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1050,16 +1050,19 @@ void RewardsServiceImpl::OnURLLoaderComplete(
             [](std::unique_ptr<std::string> response_body,
                LoadURLCallback callback, mojom::UrlResponsePtr response,
                scoped_refptr<base::SequencedTaskRunner> post_response_runner) {
-              data_decoder::JsonSanitizer::Sanitize(
+              api_request_helper::SanitizeAndParseJson(
                   *response_body,
                   base::BindOnce(
                       [](LoadURLCallback callback,
                          mojom::UrlResponsePtr response,
                          const scoped_refptr<base::SequencedTaskRunner>&
                              post_response_runner,
-                         data_decoder::JsonSanitizer::Result result) {
+                         api_request_helper::ValueOrError result) {
                         if (result.has_value()) {
-                          response->body = std::move(result).value();
+                          std::string json;
+                          base::JSONWriter::Write(std::move(result).value(),
```

Comment:
I preserved using `json_sanitizer_task_runner_` because we still has potentially heavy `base::JSONWriter::Write`

Reply:
I'll create an issue to allow us to pass the `base::Value` directly to the Rewards engine.

Once we do that, will we still need `json_sanitizer_task_runner_`?

Reply:
`json_sanitizer_task_runner_` was introduced to remove the work out from UI thread.
If we remove it now, it results in running `JSONWriter` on UI thread.

Do you sure it's okay in terms of performance? How large is the processed data?

Reply:
Let's leave `json_sanitizer_task_runner_` in this PR. I was just wondering whether we'll need it after I've removed the call to `JSONWriter::Write`.